hubble/executor/helper.py

```diff
@@ -0,0 +1,927 @@
+"""Module for helper functions for Hub API."""
+
+import functools
+import hashlib
+import io
+import json
+import logging
+import os
+import shelve
+import subprocess
+import sys
+import tarfile
+import urllib
+import uuid
+import warnings
+import zipfile
+from argparse import ArgumentParser, Namespace
+from contextlib import nullcontext
+from enum import IntEnum
+from functools import lru_cache, wraps
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+from urllib.parse import urlparse
+
+from hubble import get_token
+from hubble.executor.requirements import (
+    check_env_variable,
+    expand_env_variables,
+    get_env_variables,
+    parse_requirement,
+)
+from rich.console import Console
+
+default_logger = logging.getLogger(__name__)
+
+__resources_path__ = os.path.join(
+    os.path.dirname(sys.modules['hubble'].__file__), 'resources'
+)
+
+__unset_msg__ = '(unset)'
+
+__cache_path__ = f'{os.path.expanduser("~")}/.cache/{__package__}'
+if not Path(__cache_path__).exists():
+    Path(__cache_path__).mkdir(parents=True, exist_ok=True)
+
+
+@lru_cache()
+def get_hub_packages_dir() -> Path:
+    """Get the path of folder where the hub packages are stored
+
+    :return: the path of folder where the hub packages are stored
+    """
+    root = Path(__cache_path__)
+    hub_packages = root.joinpath('hub-package')
+
+    if not hub_packages.exists():
+        hub_packages.mkdir(parents=True, exist_ok=True)
+
+    return hub_packages
+
+
+@lru_cache()
+def get_cache_db() -> Path:
+    """Get the path of cache db of hub Executors
+
+    :return: the path of cache db of hub Executors
+    """
+    root = Path(__cache_path__)
+    cache_db = root.joinpath('disk_cache.db')
+
+    return cache_db
+
+
+@lru_cache()
+def get_download_cache_dir() -> Path:
+    """Get the path of cache folder where the downloading cache is stored
+
+    :return: the path of cache folder where the downloading cache is stored
+    """
+    cache_dir = Path(__cache_path__)
+
+    if not cache_dir.exists():
+        cache_dir.mkdir(parents=True, exist_ok=True)
+
+    return cache_dir
+
+
+class ArgNamespace:
```

Comment:
I'm thinking if we can remove this class totally. So that we can avoid `jina` as our dependency.

I checked that it's used in `__init__` function of `HubIO`, but in Jina Core, that part of code seems never been used.

Reply:
yes, you're right. `HubIO`'s `__init__` never used with `kwargs` in Jina Core.
Removed `ArgNamespace` related code.